core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -414,7 +418,8 @@ private[kafka] class Processor(val id: Int,
   }
 
   private val newConnections = new ConcurrentLinkedQueue[SocketChannel]()
-  private val inflightResponses = mutable.Map[String, RequestChannel.Response]()
+  // Visible for testing
+  private[network] val inflightResponses = mutable.Map[String, RequestChannel.Response]()
```

Comment:
If we just access this in tests, it may make sense to have a `def` that exposes just `collection.Map` so that we can be sure that the map is only modified from this class.

Reply:
Tests only need the size rather than the whole map, updated.